test/helper/e2e/actions/cloud/aws.go

```diff
@@ -280,6 +282,34 @@ func (a *AwsAction) CreatePrivateEndpoint(serviceName, privateEndpointName, regi
 		if err != nil {
 			a.t.Error(err)
 		}
+
+		timeout := 10 * time.Minute
+		start := time.Now()
+		for {
+			a.t.Log(fmt.Sprintf("deleting VPC ID %q since %v", aws.StringValue(result.VpcEndpoint.VpcEndpointId), time.Since(start)))
+
+			output, err := ec2Client.DescribeVpcEndpoints(&ec2.DescribeVpcEndpointsInput{
+				VpcEndpointIds: []*string{result.VpcEndpoint.VpcEndpointId},
+			})
+
+			var e awserr.Error
+			if (errors.As(err, &e) && e.Code() == "InvalidVpcEndpointId.NotFound") || len(output.VpcEndpoints) == 0 {
+				return
+			}
+
+			if err != nil {
+				a.t.Error(err)
+				return
+			}
+
+			if time.Since(start) > timeout {
+				a.t.Error(errors.New("timeout waiting for deletion of vpc endpoints"))
+				return
+			}
+
+			// we do know that deletion of VPC endpoints takes time
+			time.Sleep(3 * time.Second)
+		}
```

Comment:
Why do you want to wait for this confirmation? I think fire and forget(without error) is enough, no?

Reply:
no, because if we delete subnets without waiting for private endpoint deletion, it will error.